apps/ex_cucumber/lib/ex_cucumber/gherkin/traverser/feature_traverser.ex

```diff
@@ -6,27 +6,77 @@ defmodule ExCucumber.Gherkin.Traverser.Feature do
 
   alias ExGherkin.AstNdjson.Background
 
+  require Logger
+
   def run(%ExGherkin.AstNdjson.Feature{children: nil}, acc, _), do: acc
 
   def run(%ExGherkin.AstNdjson.Feature{} = f, acc, parse_tree) do
     {background, children} = background(f.children)
 
-    acc =
+    merged_ctx =
       Ctx.extra(
         acc,
         Map.merge(feature_meta(f), %{state: %{}, step_history: [], context_history: []})
       )
 
-    children
-    |> Enum.each(fn child ->
-      acc = MainTraverser.run(background, acc, parse_tree)
+    result =
+      children
+      |> Enum.reduce(merged_ctx, fn child, ctx ->
+        uodated_ctx = MainTraverser.run(background, ctx, parse_tree)
```

Comment:
```suggestion
        updated_ctx = MainTraverser.run(background, ctx, parse_tree)
```
and several other places.